chatgpt/nodes/validator.py

```diff
@@ -414,4 +416,21 @@ def resync_state(self):
 
 if __name__ == "__main__":
     vana.trace()
-    Validator().run()
+    validator = Validator()
+
+    try:
+        while True:
+            try:
+                asyncio.run(validator.run())
+            except KeyboardInterrupt:
+                print("\nKeyboard interrupt received. Stopping the validator...")
+                break
+            except Exception as e:
+                vana.logging.error(f"An error occurred: {str(e)}")
+                vana.logging.error("Restarting the validator in 30 seconds...")
+                asyncio.sleep(30)
+    except KeyboardInterrupt:
```

Comment:
Should we shut down the KeyboardInterrupt more gracefully [like here](https://github.com/vana-com/vana-dlp-chatgpt/blob/main/chatgpt/nodes/validator.py#L349)? Maybe we can even get rid of the KeyboardInterrupt there.

Reply:
Good catch, I removed KeyboardInterrupt from the `if __name__ == "__main__"` section, one in `run` method works properly, no need to duplicate that. 

I also change it a little bit to move initialization inside while loop, this cover the case when validatior starts for the first time with broken RPC / network issues.